src/entities/createPool/createPool.ts

```diff
@@ -0,0 +1,26 @@
+import { CreatePoolBase, CreatePoolInput } from './types';
+import { ValidateInputs } from './utils/validateInputs';
+import { CreatePoolWeighted } from './weighted/createPoolWeighted';
+
+export class CreatePool {
+    private readonly createPoolTypes: Record<string, CreatePoolBase> = {};
+
+    constructor() {
+        this.createPoolTypes = {
+            WEIGHTED: new CreatePoolWeighted(),
+        };
+    }
+
+    private getCreatePool(poolType: string): CreatePoolBase {
+        if (!this.createPoolTypes[poolType]) {
+            throw new Error('Unsupported pool type');
```

Comment:
I believe that logging `poolType` here could be helpful:
```suggestion
            throw new Error('Unsupported pool type: ${poolType}');
```